addons/web/static/tests/views/fields/one2many_field_tests.js

```diff
@@ -500,6 +500,48 @@ QUnit.module("Fields", (hooks) => {
         await click(target, ".o_field_many2one input");
     });
 
+    QUnit.test(
+        'O2M with buttons with attr "special" in dialog close the dialog',
+        async function (assert) {
+            patchWithCleanup(browser, {
+                setTimeout: (fn) => fn(),
+            });
+
+            serverData.models.turtle.fields.parent_id = {
+                string: "Parent",
+                type: "many2one",
+                relation: "turtle",
+            };
+            await makeView({
+                type: "form",
+                resModel: "partner",
+                serverData,
+                arch: `
+                <form>
+                    <field name="turtles">
+                        <tree>
+                            <field name="parent_id"/>
+                        </tree>
+                        <form>
+                            <field name="parent_id" domain="[('id', 'in', parent.turtles)]"/>
```

Comment:
domain not necessary either (I would even use a more basic field, like "foo", both in list and form)